### ISSUE TITLE
Add Support for Credentials

### ISSUE THREAD

**Issue by AndrewPla** [sysgoblin/PSPDQ#5]
What do you think about moving towards adding invoke-command to all commands as a wrapper and adding support for providing a pscredential object? It would extend the use cases for the module. Some commands, like `Install-PDQPackage` make use of invoke-command during the sql queries, but then just call pdqdeploy.exe directly at the end. I have updated the `Install-PDQPackage` command and added credential support. let me know what you think and if we can do this for the other commands.

**Comment by AndrewPla** [comment]
```powershell
function Install-PDQPackage {
  
    [CmdletBinding(SupportsShouldProcess = $True)]
    param (
        [Parameter(Mandatory = $true,
            ValueFromPipelineByPropertyName)]
        [string[]][alias('Name')]$Computer,

        [Parameter(Mandatory = $true,
            ParameterSetName = 'ID',
            ValueFromPipelineByPropertyName)]
        [int[]]$PackageID,

        [Parameter(Mandatory = $true,
            ParameterSetName = 'Name',
            ValueFromPipelineByPropertyName)]
        [string[]]$PackageName,

        [PSCredential]$Credential
    )

    process {

        if (!(Test-Path -Path "$($env:AppData)\pspdq\config.json")) {
            Throw "PSPDQ Configuration file not found in `"$($env:AppData)\pspdq\config.json`", please run Set-PSPDQConfig to configure module settings."
        }
        else {
            $config = Get-Content "$($env:AppData)\pspdq\config.json" | ConvertFrom-Json

            $Server = $config.Server.PDQDeployServer
            $DatabasePath = $config.DBPath.PDQDeployDB
        }

        if ($PSCmdlet.ParameterSetName -eq 'ID') {
            $Package = @()

            $PackageID | Foreach-Object {
                $i = $_
                $sql = "SELECT PackageID, Name FROM Packages WHERE PackageID = $i"

                $icmParams = @{
                    Computer     = $Server
                    ScriptBlock  = { $args[0] | sqlite3.exe $args[1] }
                    ArgumentList = $sql, $DatabasePath
                }
                if ($Credential) { $icmParams['Credential'] = $Credential }
                $packages = Invoke-Command @icmParams

                $packagesParsed = $packages | ForEach-Object {
                    $p = $_ -split '\|'
                    [PSCustomObject]@{
                        PackageID   = $p[0]
                        PackageName = $p[1]
                    }
                }

                $Package += ($packagesParsed).PackageName
            }
        }

        if ($PSCmdlet.ParameterSetName -eq 'Name') {
            $Package = @()

            $PackageName | ForEach-Object {
                try {
                    $n = $_
                    $sql = "SELECT PackageID, Name, Version FROM Packages WHERE Name LIKE '%%$n%%'"

                    $icmParams = @{
                        Computer     = $Server
                        ScriptBlock  = { $args[0] | sqlite3.exe $args[1] }
                        Argumentlist = $sql, $DatabasePath
                    }
                    if ($Credential) { $icmParams['Credential'] = $Credential }
                    $packages = Invoke-Command @icmParams

                    $packagesParsed = $packages | ForEach-Object {
                        $p = $_ -split '\|'
                        [PSCustomObject]@{
                            PackageID   = $p[0]
                            PackageName = $p[1]
                            Version     = $p[2]
                        }
                    }

                    if ($packagesParsed.Count -gt 1) {
                        Write-Output "$($packagesParsed | Out-String)"
                        $selection = Read-Host "Multiple matches, select Package ID"

                        $Package += ($packagesParsed | Where-Object PackageID -eq $selection).PackageName
                    }
                    else {
                        $Package += ($packagesParsed).PackageName
                    }
                }
                catch {
                    Throw "No packages found matching $n"
                }
            }
        }

        foreach ($pkg in $Package) {

            $icmParams = @{
                Computer     = $Server
                ScriptBlock  = { PDQDeploy.exe Deploy -Package $Using:pkg -Targets $using:Computer }
                ArgumentList = $pkg, $computer
            }
            if ($Credential) { $icmParams['Credential'] = $Credential }
            Write-Verbose "Deploying $pkg to Target $Computer"
            Invoke-Command @icmParams
        }
    }
}
```

**Comment by sysgoblin** [comment]
Merged #6
